client/src/server.rs

```diff
@@ -322,8 +398,153 @@ impl Display for ServerRequestError {
+                .is_none()
+            {
+                panic!("Cannot publish bundle twice for client. Not supported.")
+            }
+            self.state
+                .lock()
+                .await
+                .pre_key_bundles
+                .get_mut(&self.address.name().to_owned())
+                .unwrap()
+                .push(pre_key_bundle);
+            Ok(())
+        }
+
+        async fn fetch_pre_key_bundles(
+            &self,
+            service_id: &ServiceId,
+        ) -> Result<Vec<PreKeyBundle>, SignalClientError> {
+            Ok(self
+                .state
+                .lock()
+                .await
+                .pre_key_bundles
+                .get(&service_id.service_id_string())
+                .unwrap()
+                .clone())
+        }
+
+        async fn register_client(
+            &self,
+            phone_number: String,
+            password: String,
+            registration_request: common::web_api::RegistrationRequest,
+            session: Option<&super::VerifiedSession>,
+        ) -> Result<common::web_api::RegistrationResponse, SignalClientError> {
+            todo!()
+        }
+
+        async fn send_msg(
+            &mut self,
+            messages: SignalMessages,
+            service_id: &ServiceId,
+        ) -> Result<(), SignalClientError> {
+            for message in messages.messages {
+                let envelope = Envelope::builder()
+                    .r#type(message.r#type)
+                    .content(message.content.as_bytes().to_vec())
+                    .source_service_id(self.address.name().to_owned())
+                    .source_device(self.address.device_id().into())
+                    .build();
+                self.state
+                    .lock()
+                    .await
+                    .message_queues
+                    .entry(ProtocolAddress::new(
+                        service_id.service_id_string(),
+                        message.destination_device_id.into(),
+                    ))
+                    .or_insert(vec![])
+                    .push(envelope);
+            }
+            Ok(())
+        }
+
+        async fn get_message(&mut self) -> Option<common::signalservice::Envelope> {
+            self.state
+                .lock()
+                .await
+                .message_queues
+                .get_mut(&self.address)
+                .unwrap()
+                .pop()
+        }
+
+        fn create_auth_header(
+            &mut self,
+            aci: libsignal_core::Aci,
+            password: &str,
+            device_id: libsignal_core::DeviceId,
+        ) -> () {
+            todo!()
+        }
+
+        fn new() -> Self {
+            todo!()
+        }
+    }
+
+    impl MockBackend {
+        pub fn new(address: ProtocolAddress, state: Arc<Mutex<MockBackendState>>) -> Self {
+            Self { address, state }
+        }
+    }
+}
+*/
```

Comment:
Do we need this?

Reply:
Venter på alex Keymanager create_key_bundle